examples/.experimental/next-app-dir/src/trpc/server-invoker.ts

```diff
@@ -21,15 +21,9 @@ export const api = experimental_createTRPCNextAppDirServer<typeof appRouter>({
           revalidate: 5,
           router: appRouter,
           transformer: superjson,
-          createContext: async () => {
-            return {
-              session: await auth(),
-              headers: {
-                cookie: cookies().toString(),
-                'x-trpc-source': 'rsc-invoke',
-              },
-            };
-          },
+          // include the user id in the cache key
+          cacheContext: (ctx) => [ctx.session?.user.id],
+          createContext,
```

Comment:
we don't need it on the link then right? if it's required higher up?

Reply:
you're right, we have to hoist it down somehow. couldn't find a fast way to do that.

Reply:
iirc it'll be on the runtime